### DISCUSSION TITLE
--disable_audio_effects

### DISCUSSION THREAD

**Discussion by Oldblob** [cktlco/crew-chief-autovoicepack#41] [Troubleshooting] [answered]
Hi,

first, thanks for your work. I used this to only generate phrases for the iracing spotter, and it works like a charm, until I want to use the --disable_audio_effects switch. Then no generated audio file is passing any validity checks, it's usually 0.0 maybe 0.04.

I tried instead not using the switch and disable the parts of the audio effects I dont want (overdrive, equilizer, keeping silence cut) in the generate_voice_pack.py,  but the outcome is the same. 

I mounted the generate_voice_pack.py directly and restarted the docker each time. I checked with vim after starting the docker, if it is my generate_voice_pack.py file, and it is.

I want to disable the effects, because iracing is putting its own effects on anyway, and downsamples the files to 8000Hz.

thanks again.

**Comment by cktlco** [comment, marked as answer]
Hi, thanks for reaching out, it's excellent to see that you're so far, and great troubleshooting attempts.

Near term, let's try either `--simple_validity_check` or `--xtts_integrity_threshold 0.0` to see if that helps.

The issue might be that the "good audio" classifier model either incorrectly condemns files without the audio effects (ie, it's overfit to examples including the effects), or possibly something about the input voice is throwing off its abilities. That first param should fall back to the classic sanity check, and the second will keep the ML model check but with a very low threshold.

It's possible it would help for me to retrain the [xtts-integrity](https://github.com/cktlco/xtts-integrity) model with more examples (without audio effects) which might help, and I can consider doing that once we can get past your issue.

**Comment by Oldblob** [comment]
Thanks for the reply.

--simple_validity_check did work, but gave me excess silence warnings with effects disable switch. I opted to disable the unwanted effects in the phyton file keeping the normalize and cut silence, also raised silence threshold to 0.6 and fiddling with speed right now.

Important part is the clean audio i am getting :D 

Thanks again.

**Reply by cktlco** [reply]
Excellent! Thanks for the update and great job for pioneering a new use case and being willing to tinker with the tooling. I'll keep this in mind.

**Comment by Oldblob** [comment]
One question. How is to get and american, english or irish accent out of the voice? Only with the recordings? Or are there language parameters I could use?

**Reply by cktlco** [reply]
It can be tricky. The primary contributor is the accent/language used in the original voice recordings. Otherwise, you can try to tweak the exact text sent to the text-to-speech model ("Looees" instead of "Luis") using the "text_for_tts" column, maybe including authentic regional jargon. 

In general, I've had success with English-only voices speaking German/Portuguese and other multilingual text, and using an input voice with a strong accent with English text also works (the "Shannon" voicepack is an Irish example). Certain voices/accents (British, Irish, Indian) seem to work better than others (Japanese).

**Reply by Oldblob** [reply]
Thanks for the pointers.
